### PR TITLE
Refine queue controller

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
1. When podgroup.spec.queue not specified, use `default` by default, which is consistent with scheduler.

1. Do not treat non podgroups as an error, since it is common.

### CODE REVIEW

**Comment by k82cn** [comment]
IMO, we should keep Queue event handler here; we should not let queue events dependent on podgroups events.

**Comment by hzxuzhonghu** [comment]
We should only update queue status until there are podgroups for it.

**Comment by TommyLike** [comment]
> We should only update queue status until there are podgroups for it.

@hzxuzhonghu Other than the discussion point here, is it possible that we could update the queue to the incorrect status when the controller restarted that only part of the PG has been cached locally?

**Comment by k82cn** [comment]
> We should only update queue status until there are podgroups for it.

Let's keep event handler there; that's really wired that Queue's handler dependent on PodGroup's event :(

**Comment by hzxuzhonghu** [comment]
Then i think this part need a refactor

**Comment by k82cn** [comment]
> Then i think this part need a refactor

Please find another solution for that.

File: pkg/controllers/queue/queue_controller.go [hunk 1]
```diff
@@ -279,12 +280,21 @@ func (c *Controller) deletePodGroup(obj interface{}) {
...
 
-	delete(c.podGroups[pg.Spec.Queue], key)
+	delete(c.podGroups[queue], key)
 
-	c.queue.Add(pg.Spec.Queue)
+	c.queue.Add(queue)
+}
+
+func getPogGroupQueue(pg *schedulingv1alpha2.PodGroup) string {
+	if pg.Spec.Queue == "" {
```
Comment (k82cn) [inline pkg/controllers/queue/queue_controller.go]: prefer to handle this in admission controller.

Comment (hzxuzhonghu) [reply pkg/controllers/queue/queue_controller.go]: We can mutate in admission webhook, but keep in mind admission is an optional component.

We do not have admission for pg now, it is a big cost just for a default value set. Anyway if you really insist on that,   crd definition can set default value.

**Pull request closed without merging**